Lombiq.Tests.UI/Extensions/VisualVerificationUITestContextExtensions.cs

```diff
@@ -400,22 +399,34 @@ public static class VisualVerificationUITestContextExtensions
         // We take a screenshot of the element area. This will be compared to a baseline image.
         using var elementImageOriginal = context.TakeElementScreenshot(element).ShouldNotBeNull();
 
-        // Checking the size of captured image.
-        try
-        {
-            elementImageOriginal.Width
-                .ShouldBeGreaterThanOrEqualTo(cropRegion.Left + cropRegion.Width);
-            elementImageOriginal.Height
-                .ShouldBeGreaterThanOrEqualTo(cropRegion.Top + cropRegion.Height);
-        }
-        catch
-        {
-            if (approvedContext != null)
+        var originalElementScreenshotFileName =
+            new[]
             {
-                context.SaveSuggestedImage(element, approvedContext.BaselineImagePath, approvedContext.BaselineFileName);
+                configuration.FileNamePrefix,
+                VisualVerificationMatchNames.ElementImageFileName,
+                configuration.FileNameSuffix,
             }
+            .JoinNotNullOrEmpty("-");
 
-            throw;
+        // Checking the dimensions of captured image. This needs to happen before any other comparisons, because that
+        // can only be done on images with the same dimensions.
+        var cropWidth = cropRegion.Left + cropRegion.Width;
+        var cropHeight = cropRegion.Top + cropRegion.Height;
+        if (elementImageOriginal.Width < cropWidth || elementImageOriginal.Height < cropHeight)
+        {
+            var cropRegionName = regionOfInterest == null ? "baseline image" : "selected region of interest";
+            var message = $"The dimensions of the captured element ({elementImageOriginal.Width.ToTechnicalString()}" +
+                $"px x {elementImageOriginal.Height.ToTechnicalString()}px) are smaller than the dimensions of the " +
+                $"{cropRegionName} ({cropWidth.ToTechnicalString()}px x {cropHeight.ToTechnicalString()}px). This " +
+                "can happen if due to a change in the app the captured element got smaller than before, or if the " +
+                $"{cropRegionName} is mistakenly too large.";
+
+            message += " The suggested baseline image with a screenshot of the captured element was saved to the " +
+                "failure dump. Compare this with the original image used by the test and if suitable, use it as the " +
+                "baseline going forward.";
```

Comment:
Why `message +=` is required here?
```suggestion
            var message = $"The dimensions of the captured element ({elementImageOriginal.Width.ToTechnicalString()}" +
                $"px x {elementImageOriginal.Height.ToTechnicalString()}px) are smaller than the dimensions of the " +
                $"{cropRegionName} ({cropWidth.ToTechnicalString()}px x {cropHeight.ToTechnicalString()}px). This " +
                "can happen if due to a change in the app the captured element got smaller than before, or if the " +
                $"{cropRegionName} is mistakenly too large." +
                " The suggested baseline image with a screenshot of the captured element was saved to the " +
                "failure dump. Compare this with the original image used by the test and if suitable, use it as the " +
                "baseline going forward.";
```

Reply:
It isn't, just a leftover from the previous version.